File: api/src/notify/sendEvent.ts
import FormData from "form-data";
import axios from "axios";
import { logger } from "../logger";

export const sendEvent = async (
  ratio: number,
  duration: number,
  time: string,
  mepasRsam: number,
  id: string
) => {
  const form = new FormData();
  const text = `Terjadi gempa:\nWaktu: ${time} WIB\nRSAM: ${mepasRsam}\nDurasi: ${duration} detik\nRatio Max t <= 30: ${ratio}\n\n${id}`;
  form.append("chat_id", "-1002026839953");
  form.append("text", text);
  form.append("parse_mode", "Markdown");

  try {
      const { data } = await axios.post(
        `https://api.telegram.org/bot${process.env.BOT_TOKEN}/sendMessage`,
        form
      );
      logger.info("sent event to telegram: ", data);
  } catch (error) {
    logger.info("failed to send event to telegram: ", error);
  }
};

File: api-py/sendEvent.py
import os
import aiohttp
from dotenv import load_dotenv

load_dotenv()
chat_id = os.getenv('CHAT_ID')
bot_token = os.getenv('BOT_TOKEN')

async def sendEvent(date: str, ratio: float, mepasRsam: int, duration: int, output: str):
    photo_path = f'{output}'
    time = date
    caption = (
        f'Terjadi gempa:\n'
        f'Waktu: {time} WIB\n'
        f'RSAM: {mepasRsam}\n'
        f'Durasi: {duration} detik\n'
        f'Ratio: {ratio}'
    )
    
    url = f'https://api.telegram.org/bot{bot_token}/sendPhoto'
    
    async with aiohttp.ClientSession() as session:
        with open(photo_path, 'rb') as photo:
            form = aiohttp.FormData()
            form.add_field('photo', photo)
            form.add_field('chat_id', chat_id)
            form.add_field('caption', caption)
            form.add_field('parse_mode', 'Markdown')
            
            async with session.post(url, data=form) as response:
                return await response.json()